rocAL/source/pipeline/master_graph.cpp

```diff
@@ -981,9 +981,10 @@ void MasterGraph::output_routine() {
             _ring_buffer.push();  // The data and metadata is now stored in output the ring_buffer, increases it's level by 1
         }
     } catch (const std::exception &e) {
-        ERR("Exception thrown in the process routine: " + STR(e.what()) + STR("\n"));
         _processing = false;
         _ring_buffer.release_all_blocked_calls();
+        release();
+        THROW("Exception thrown in the process routine: " + STR(e.what()) + STR("\n"));
```

Comment:
Instead of raising exception here, in the run function, we need to check if the processing was succes or not. Then return appropriate error code from there.

Reply:
Done

Reply:
I meant in the run() function. If we throw exception, we need to make sure it is caught somewhere and gracefully exit with error code

Reply:
Done